host/libhackrf/src/hackrf.c

```diff
@@ -113,6 +118,7 @@ struct hackrf_device {
 	hackrf_sample_block_cb_fn callback;
 	volatile bool transfer_thread_started; /* volatile shared between threads (read only) */
 	pthread_t transfer_thread;
+	volatile hackrf_transfer_direction transfer_direction; /* volatile shared between threads (read only) */
```

Comment:
I assume this would require a libhackrf major version bump?

Reply:
`hackrf_device` is an opaque struct that is allocated by libhackrf and a pointer returned to the user via `hackrf_open(hackrf_device** device)`. This shouldn't change the ABI or require a version bump.